resources/assets/js/photoAlbum.js

```diff
@@ -0,0 +1,239 @@
+
+    //check if albumname, description and date exists.
+    if(albumNameInput  && albumDescriptionInput && captureDateInput){
+        //check if input of the albumname,description and date are correct.
+        if(albumNameInput.checkValidity() && albumDescriptionInput.checkValidity() && captureDateInput.checkValidity()){
+            albumName = albumNameInput.value;
+            albumDescription = albumDescriptionInput.value;
+            captureDate = captureDateInput.value;
+            var justPhoto = (albumName === undefined || albumDescription === undefined);
+    
+            //check if albumname, description is not empty and not exceed 250 characters.
+            if(!justPhoto && (albumName.length >= 250 || albumDescription.length >= 250)) {
+                alert("The album name and description have to consist of less than 250 characters.");
+                return;
+            }
+        } else return;
+    }
+
+    //check if selectedfile input is valid (not null)
+    if(fileInput.checkValidity()){
+        files = Array.from(fileInput.files);
+        numberOfPhotos = files.length;
+        updateProgressBar(); //show progressbar
+        processPhotos(); //start processing photos
+    } else alert("Selecteer een aantal foto's om toe te voegen."); 
+}
+
+
+var fileIndex = 0; //fileIndex of the files array, needs to be global because its a recursive function.
+//recusrive function the process photos.
+function processPhotos(){
+    ChangebuttonState("Uploading photos...");
+    resizeThumbnail(files[fileIndex]).then(function({thumbnail, file}){
+        downscalePhoto(file).then(function({photo, file}){
+            if (currentPhotoAlbum != undefined) { //if currentAlbum is undefined then it means that we are adding an album.
+                addPhotoToAlbum(thumbnail, photo, fileIndex).then(function(index){
+                    if(index >= (files.length -1)){ //if returned index equals files length then last photo has uploaded. refresh page
+                        location.reload();
+                    } else{
+                        updateProgressBar();
+                    }    
+                });
+                fileIndex++; //increase Fileindex for next recurse
+                processPhotos();
+            } else{
+                addAlbum(thumbnail, photo, albumName, albumDescription, captureDate, fileIndex).then(function(index){
+                    if(index >= (files.length -1)){ //if returned index equals files length then last photo has uploaded. refresh page
+                        location.reload();
+                    } else{
+                        updateProgressBar();
+                        fileIndex++; //increase Fileindex for next recurse
+                        processPhotos();
+                    }
+                });
+            }     
+        }) 
+    });
+    
+}
+
+
+//Downscales the original photo to a smaller size
+//file: Photo to downscale
+function downscalePhoto(file) {
```

Comment:
Persoonlijk denk ik dat de jpeg compressie op het moment wel erg hoog is. Er gaat best wel een hoop kwaliteit verloren. Is het een idee om het compressie level in bijvoorbeeld de `.env` op te slaan?